File: Warehousing.Data/Entities/SubCategory.cs
namespace Warehousing.Data.Entities
{
    public class SubCategory
    {
        public int Id { get; set; }
        public string? NameEn { get; set; } = string.Empty;
        public string? NameAr { get; set; } = string.Empty;
        public string? Description { get; set; } = string.Empty;
        public bool IsActive { get; set; } = true;

        public int CategoryId { get; set; }
        public virtual Category Category { get; set; } = null!;
        public virtual ICollection<Product> Products { get; set; } = new List<Product>();
    }
}

File: Warehousing.UI/src/app/admin/models/SubCategory.ts
import { Category } from "./category";

export interface SubCategory {
    id: number;
    nameEn: string | null;
    nameAr: string | null;
    description: string | null;
    imagePath: string | null;
    isActive: boolean;
    categoryId: number;
    category: Category;
    createdAt: string | null;
    createdBy: string | null;
    updatedAt: string | null;
    updatedBy: string | null;
    // Removed products collection to match API changes (prevents circular reference)
}